app/controllers/session_controller.rb

```diff
@@ -107,7 +108,6 @@ def sso_provider(payload = nil, confirmed_2fa_during_login = false)
 
     def become
       raise Discourse::InvalidAccess if Rails.env.production?
-      raise Discourse::ReadOnly if @readonly_mode
```

Comment:
Removed that line to make it a better DX. So we can use this `/session/XXXX/become` route in development` even though the site is in readonly mode.